comp/core/autodiscovery/autodiscoveryimpl/autoconfig_test.go

```diff
@@ -500,3 +513,12 @@ func TestProcessClusterCheckConfigWithSecrets(t *testing.T) {
 	newCheckID := checkid.BuildID(resolved.Name, resolved.FastDigest(), resolved.Instances[0], resolved.InitConfig)
 	assert.Equal(t, originalCheckID, ac.GetIDOfCheckWithEncryptedSecrets(newCheckID))
 }
+
+type Deps struct {
+	fx.In
+	WMeta workloadmeta.Component
+}
+
+func createDeps(t *testing.T) Deps {
```

Comment:
maybe add comments to a bit explain what it does inside ? I feel that when we are not used with FX it can be a bit hard to understand everything ? WDYT ?